### PR TITLE
added Doors Data and its properties

### PR DESCRIPTION
**Pull request by krishnabangalore**
jira issue: wp-614

Doors Data for Vehicle API and its Properties pushed

Habib/Andreas: This seems to be clean I will be deleting the previous request

### CODE REVIEW

**Comment by krishnabangalore** [comment]
@andreasbotsikas  / @habibvirji : can you please merge the changes since I need this doors data in the simulation part to see how the window work.

I have fixed the Indentation and the commented case. anything not fixed let me know

**Comment by krishnabangalore** [comment]
Can any one of  you merge this pull request Please??

File: webinos/core/api/vehicle/contrib/vb-sim/simulator/vehicle.html [hunk 1]
```diff
@@ -54,9 +55,9 @@
             <option value='5'>5</option>
             <option value='6'>6</option>
           </select>
-        </p>
-        	<h3>Tripcomputer</h3>
-
+         </p>
+        	
+        <h3>Tripcomputer</h3>
```
Comment (jplyle) [inline webinos/core/api/vehicle/contrib/vb-sim/simulator/vehicle.html:60]: Should this be "Trip computer" ?

File: webinos/core/api/vehicle/contrib/vb-sim/simulator/vehicle.html [hunk 2]
```diff
@@ -86,8 +87,8 @@
           <input class='input' type='text' value='33.3' id='tc-d'>
         </p>
         <input type='button' id='setTripComputer' value="Set Tripcomputer" />
-        	<h3>Parksensors Front</h3>
-
+        	
+        <h3>Parksensors Front</h3>
```
Comment (jplyle) [inline webinos/core/api/vehicle/contrib/vb-sim/simulator/vehicle.html:91]: "Front parking sensors" ? ... and same for rear, too.

File: webinos/core/api/vehicle/contrib/vb-sim/simulator/vehicle.html [hunk 3]
```diff
@@ -189,7 +190,51 @@
...
+        <p>
+          <label class='select'>trunkDeck:</label>
+          <select id="dd-td">
+            <option value='true'>Opened</option>
+            <option value='false' selected>Closed</option>
+          </select>
+        </p>
+        <input type='button' id='setDoorsData' value="Set Doors Data" />
+       
+   <!--   <p>
```
Comment (jplyle) [inline webinos/core/api/vehicle/contrib/vb-sim/simulator/vehicle.html:232]: There's not much point committing a comment ... might be worth removing?

File: webinos/core/api/vehicle/contrib/vb-sim/vs.js [hunk 4]
```diff
@@ -66,6 +67,28 @@
...
+/*-------Relevant Objects for Doors Data------*/
+
+     DoorsDataEvent = function (ddData) {
+        this.initDoorsDataEvent(ddData);
+    }
+    DoorsDataEvent.prototype = new WDomEvent();
+    DoorsDataEvent.prototype.constructor = ShiftEvent;
+    DoorsDataEvent.parent = WDomEvent.prototype; // our "super" property
+    DoorsDataEvent.prototype.initDoorsDataEvent = function (ddData) {
+        this.driver = ddData.d;
```
Comment (jplyle) [inline webinos/core/api/vehicle/contrib/vb-sim/vs.js:79]: Is any input validation required here?  e.g., if the doors data is invalid / missing?

**Comment by jplyle** [comment]
The issue you refer to appears to be closed - http://jira.webinos.org/browse/wp-614 - is that in anticipation of this PR?

Also, if either the jira issue OR this PR could describe what this issue addresses, that would be very helpful for clueless reader such as myself.

**Comment by jplyle** [comment]
The Travis build looks good: https://travis-ci.org/webinos/Webinos-Platform/builds/6441216 - but unfortunately (in this case) it doesn't really cover any tests for this item.

@krishnabangalore - if you can explain how this item can be tested, I can have a look.

**Comment by krishnabangalore** [comment]
Yes, It's the same Jira issue WP-614 but it is not yet merged. I'm waiting for this one to be merged which is stopping me from adding other Vehicle Interfaces and its properties. 

Jira issue: WP-614 adds Doors Properties as specified in the Specs whether a door is closed or opened and shows the output in the testbed when you click on the Get Data to show you whether it is true or false.

You can test it with the Simulator which is on localhost:9898/simulator/vehicle.html If you change the door settings to  closed or open it should return a value in testbed false or true.

File: webinos/core/api/vehicle/contrib/vb-sim/simulator/vehicle.html (hunk 1 above)
Comment (krishnabangalore) [inline webinos/core/api/vehicle/contrib/vb-sim/simulator/vehicle.html:60]: I can change it if think it needs to be changed in the next PR when I add the windows Data. but this shouldn't matter much for the naming concerns I will have a look and change it

File: webinos/core/api/vehicle/contrib/vb-sim/simulator/vehicle.html (hunk 3 above)
Comment (krishnabangalore) [inline webinos/core/api/vehicle/contrib/vb-sim/simulator/vehicle.html:232]: Can I do the changes in the next PR ?? -(Park sensors) The naming stuff and the cleaning part(comments) and check the identation if anything wrong

**Comment by jplyle** [comment]
Ok, I'll check it out.  If it looks good, I'll merge it.  Please address my comments in the next PR ASAP, and please add some more information to the description of this pull request and the jira issue.  Thanks.

**Comment by jplyle** [comment]
The 'get' and 'set doors data seems to work.  Should the event listener be working?  E.g., if I click 'add eventlistener' and then change doors data, should the data automatically update?  Because it doesn't...

**Comment by jplyle** [comment]
As an unrelated bug, the 'Trip Computer' link launches a trip computer, but if you then kill it, the PZP ends up being sent endless invalid update messages.  I'm not sure why that's happening, but it looks like a bug.

**Comment by krishnabangalore** [comment]
Yes, if we have added event listener and if we change the simulator interface settings then it should automatically update, I have this add event listener stuff in mind which I need to ask Simon regarding this which needs a fix for some of the interfaces.

I haven't checked the Trip Computer bug if you can open a jira issue will look in to it.

**Comment by jplyle** [comment]
So: Is this PR broken (because the event listener doesn't do anything for doors data) or is it OK, because this fix will happen in a future PR?

**Comment by krishnabangalore** [comment]
The fix will happen in the next PR.

**Comment by jplyle** [comment]
Krishna - be aware that these changes will not automatically propagate to the modularisation of webinos.

**Comment by krishnabangalore** [comment]
Thanks. OK

**Pull request merged**